### DISCUSSION TITLE
Roads with surface=compacted

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#521] [Q&A] [answered]
Unfortunately, my motorcycle hates gravel roads, so I have to turn off the "Use gravel roads" switch. Today I had an unpleasant surprise. Since the road looked beautiful on the map, through the forest and among swamps, Cruiser took me through it. But, at some point the asphalt ended – gravel started...

I checked that in OSM the road is tagged as `surface=compacted`. The OSM Wiki describes it, perhaps in many cases exaggeratedly, as unpaved. Enduro definitely loves it as much as gravel 🙂

Maybe it wouldn't be a stretch to include such roads in the same exclusion switch as gravel roads (because probably not for unpaved ones)?

**Comment by devemux86** [comment]
@Ciutnik thanks for the report.

This tag exists in BRouter database, but it does not exist in the profile rules.

Can you post an OpenStreetMap way with this tag to make tests?

**Reply by Ciutnik** [reply]
I mean this way:

https://www.openstreetmap.org/way/188644995#map=14/52.0266/21.4268

**Reply by devemux86** [reply]
@Ciutnik thanks for the sample.

So let's take a route example on [BRouter website](https://brouter.de/brouter-web/#map=13/52.0323/21.4184/standard&lonlats=21.435891,52.039161;21.414886,52.010373&profile=car-fast).

Changing line 188 in Cruiser driving profile (or line 107 in BRouter's car profiles) it could work:
`switch surface=fine_gravel|compacted ( switch avoid_unpaved 3 15 )`

**Reply by Ciutnik** [reply]
Does this mean that it would be possible to eliminate these roads using the "Use gravel roads = off" switch.

It's a really beautiful place, but I wouldn't want to go this route again 🙂

**Reply by devemux86** [reply]
@Ciutnik We need to make more tests and with these options,
this was just a first quick change in the profile to see the result.

Indeed we should influence such roads with the options unpaved or gravel.

**Reply by Ciutnik** [reply]
I think (in this case) it's more gravel than unpaved:
![Screenshot_20240502_215115_Insta360](https://github.com/devemux86/cruiser/assets/122647782/fc3661ac-3453-492c-a2f1-dfd09e7761dc)
_Picture from my 360° camera_ 🙂

**Reply by devemux86** [reply]
The gravel option is used for the highway=track roads.
The surface tag is handled by the unpaved option.

We could probably combine the 2 options into 1 and just select unpaved on / off?

**Reply by Ciutnik** [reply]
> We could probably combine the 2 options into 1 and just select unpaved on / off?

Are you thinking about a combination of gravel and unpaved? Users probably treat it as a set and either turn both on or off at the same time. But this may be misleading, because paved gravel is different from unpaved sand. Unless you just mean enabling only "compacted" in the "Avoid unpaved" toggle.

And, by the way, an observation related to this: We have two switches: "Avoid unpaved roads" and "Use gravel roads". Wouldn't it be better if both were "Avoid..." or "Use..."? 🙂

**Reply by devemux86** [reply]
If we combine the 2 options, then the `highway=track` and the `surface=...` tags will be handled with 1 option.

About the naming, avoid unpaved roads is a common negative option (avoidance).
Use gravel roads / tracks is used as a positive option (preference) in other routers, e.g. [Valhalla](https://valhalla.openstreetmap.de/).
(it can become an avoidance if we reverse the rule)

**Reply by devemux86** [reply]
If we combine the 2 options, then when avoid unpaved is off it can use all roads, even:
- [highway=track](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dtrack)
- [tracktype=](https://wiki.openstreetmap.org/wiki/Key:tracktype) (all values)

These are now filtered by the gravel roads option. So I do not know if we really want to allow this?

**Reply by Ciutnik** [reply]
> it can become an avoidance if we reverse the rule

This is exactly what I had in mind - to standardize the behavior of the switches (sliders) within one options window. All of them concern "avoidance" and only one (actually "Ignore closed roads" too) "preferring". But actually, I didn't think about the nomenclature used in other routers 🙂

**Reply by devemux86** [reply]
Perhaps the "Use gravel roads" is not the right name for "highway=track" roads?

Other routers (like Valhalla) simply name it "Use tracks", without mentioning the surface.

**Comment by devemux86** [comment]
I updated the Cruiser driving profile.

The `surface=compacted` can now be handled like other surfaces with the option to avoid unpaved.
- https://github.com/devemux86/cruiser/discussions/272

**Reply by devemux86** [reply]
@Ciutnik let's keep it open, as it will be included in an update after the next release.

**Comment by Ciutnik** [comment]
You solved this perfectly in Cruiser 4 👍

With avoidance enabled, even if I place shaping points near the beginning and end of this road, Cruiser will not guide me this way. But if I am stubborn and decisive, and place the shaping point somewhere in the middle of this path - Cruiser will respect my choice. Thanks! This is exactly what I expected 🙂

<img src="https://github.com/devemux86/cruiser/assets/122647782/77324f7d-f98d-4f16-a56f-49849bbf726b" width="49%"/> <img src="https://github.com/devemux86/cruiser/assets/122647782/0e98919b-7d02-4340-bd18-0c7110c0d2ab" width="49%"/>

**Reply by devemux86** [reply]
The [above improvement](https://github.com/devemux86/cruiser/discussions/521#discussioncomment-9303052) is not included in Cruiser 4 (already published).

Cruiser 4 contains another improvement in unpaved roads that also helps:
- https://github.com/devemux86/cruiser/discussions/504

**Reply by Ciutnik** [reply]
Well, I fooled around a bit 😁

But anyway, as you can see from the pictures, SOMETHING helped here, and I still thank you for it 🙂

**Comment by devemux86** [comment, marked as answer]
Roads with surface=compacted is included in [Cruiser 4.0.1](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
